conda_build/variants.py

```diff
@@ -634,6 +634,14 @@ def get_vars(variants, loop_only=False):
     return loop_vars
 
 
+def is_loop_var(var, variants):
+    if var in {'pin_run_as_build', 'zip_keys', 'ignore_version'}:
+        return False
+    if var in ensure_list(variants[0].get('extend_keys')):
```

Comment:
Mind moving out `variants[0]` out so it can be used in this condition and the `any` call below?

BTW, could you elaborate the need to just check the first variant? A few code comments would go a long way for future maintenance :)